inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -616,11 +616,26 @@ def normalize_journal_title_entry(publication_info):
         RecordMetadata.json['_collections'].op('?')('Journals')).filter(
         cast(RecordMetadata.json['short_title'], String) == type_coerce(normalized_title, JSON))
     result = db.session.execute(ref_query).fetchone()
-    journal_record = result.records_metadata_json['self'] if result else None
+    journal_data = result.records_metadata_json if result else None
+
+    if not journal_data:
+        return
+
+    journal_record = journal_data['self']
 
     if journal_record:
         publication_info['journal_record'] = journal_record
 
+    if journal_data.get('inspire_categories'):
+        current_categories = obj.extra_data.get('journal_inspire_categories', [])
+        current_categories.extend(journal_data['inspire_categories'])
+        obj.extra_data['journal_inspire_categories'] = current_categories
```

Comment:
didn't realize first, but this is a one-liner using `setdefault`
```suggestion
        obj.extra_data.setdefault('journal_inspire_categories', []).extend(journal_data['inspire_categories'])
```